network_simulator/link.py

```diff
@@ -50,6 +104,21 @@ def __eq__(self, other):
         return isinstance(other, self.__class__) and \
             self.__dict__ == other.__dict__
 
+    def get_packet(self):
```

Comment:
Getters really aren't necessary for most Python stuff, since Python doesn't have private variables. This isn't a huge style issue, it'll just make parts of the codebase inconsistent.